packages/account/src/providers/resource-cache.ts

```diff
@@ -73,7 +111,16 @@ export class ResourceCache {
     return false;
   }
 
-  clear() {
-    cache.clear();
+  clear(): void {
+    this.getActiveCache().clear();
+  }
+
+  reset() {
+    this.clear();
+    this.globalCache.reset();
+  }
```

Comment:
Will the `GlobalCache` be used exclusively by the `ResourceCache` class? I’m asking to ensure that resetting it won’t inadvertently clear data we still depend on.

Reply:
Yes, this seems wrong.

If one instantiates two providers, one `local` and one `global,` and then I call `local.reset()`, it will call the `global.reset()` and inadvertently clear the global cache.

Additionally, if one instantiates the `local` provider from a place/function that goes out of scope before the `global`, the latter will also be reset. This will happen because you chose to use `Symbol.dispose` and call reset from inside it.

https://github.com/FuelLabs/fuels-ts/blob/350dfd01107d7e9ff5cc919d37f3c3659a28f303/packages/account/src/providers/resource-cache.ts#L123-L125

This case is even worse since users will never manually call `.reset()` and will still have their cache wiped out.

Reply:
When I had designed this the [original intention was for consumers](https://github.com/FuelLabs/fuels-ts/pull/3454/files#diff-4dc05f389c3f0a28d095f41e799a4c9c4e024cb6a39f0ce935fe8e03da453984R73) to use exclusively a global cache strategy or an instance specific cache strategy, not a combination of both, but you're right in that this isn't programmatically enforced so the above scenario would happen.